WordPress/Classes/ViewRelated/Blog/Blog Details/SoTW 2023/SOTWCardView.swift

```diff
@@ -0,0 +1,176 @@
+    static let livestreamURLString = "https://wordpress.org/state-of-the-word/?utm_source=mobile&utm_medium=appnudge&utm_campaign=sotw2023"
+}
+
+// MARK: - UITableViewCell Wrapper
+
+class SotWTableViewCell: UITableViewCell {
+
+    private lazy var cardView: SotWCardView = {
+        let cardView = SotWCardView()
+        cardView.translatesAutoresizingMaskIntoConstraints = false
+        cardView.didTapButton = {
+            WPAnalytics.track(.sotw2023NudgePostEventCardCTATapped)
+            guard let destinationURL = URL(string: SotWConstants.livestreamURLString) else {
+                return
+            }
+            UIApplication.shared.open(destinationURL)
+        }
+
+        return cardView
+    }()
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        setupView()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupView() {
+        contentView.addSubview(cardView)
+        contentView.pinSubviewToAllEdges(cardView, priority: .defaultHigh)
+        WPAnalytics.track(.sotw2023NudgePostEventCardShown)
+    }
+
+    @objc func configure(onCardHidden: (() -> Void)?) {
+        cardView.didHideCard = onCardHidden
+    }
+}
+
+// MARK: - BlogDetailsViewController Section
+
+extension BlogDetailsViewController {
+
+    @objc func sotw2023SectionViewModel() -> BlogDetailsSection {
+        let row = BlogDetailsRow()
+        row.callback = {}
+        let section = BlogDetailsSection(title: nil,
+                                         rows: [row],
+                                         footerTitle: nil,
+                                         category: .sotW2023Card)
+        return section
+    }
+
+    @objc func shouldShowSotW2023Card() -> Bool {
+        guard AppConfiguration.isWordPress && RemoteFeatureFlag.wordPressSotWCard.enabled() else {
+            return false
+        }
+
+        // ensure that the card is not hidden.
+        let repository = UserPersistentStoreFactory.instance()
+        let cardIsHidden = repository.bool(forKey: SotWConstants.hideCardPreferenceKey)
+
+        // ensure that the device language is in English.
+        let usesEnglish = WordPressComLanguageDatabase().deviceLanguageSlugString() == "en"
+
+        // ensure that the card is not displayed before Dec. 11, 2023 where the event takes place.
+        let dateComponents = Date().dateAndTimeComponents()
+        let isPostEvent = {
+            guard let day = dateComponents.day,
+                  let month = dateComponents.month,
+                  let year = dateComponents.year else {
+                return false
+            }
+            return year >= 2024 || (year < 2024 && month == 12 && day > 11)
+        }()
+
+        return !cardIsHidden && usesEnglish && isPostEvent
+    }
```

Comment:
Is the plan to use the remote feature flag to hide this card after we no longer want to show it?

Reply:
Yes, that's correct. @thomashorta and I discussed if it's better to have the flag disabled by default and wait for the remote value to turn it on, or have it enabled by default.

Considering we want to reach as many WP app users, enabling this by default from the client side should optimistically reach more users. Then, we'll only need to deploy the remote feature flag once to disable the card.